gcloud-java-examples/src/main/java/com/google/cloud/examples/datastore/DatastoreExample.java

```diff
@@ -131,25 +158,37 @@ public void run(Transaction tx, Key userKey, String... args) {
     }
 
     @Override
-    public String getRequiredParams() {
-      return "";
+    Void parse(String... args) throws Exception {
+      return null;
     }
   }
 
-  private static class AddAction implements DatastoreAction {
+  /**
+   * This class adds a comment for a user. If the user does not exist its entity is created.
+   */
+  private static class AddCommentAction extends DatastoreAction<String> {
     @Override
-    public void run(Transaction tx, Key userKey, String... args) {
+    public void run(Transaction tx, Key userKey, String content) {
       Entity user = tx.get(userKey);
       if (user == null) {
         System.out.println("Adding a new user.");
-        user = Entity.builder(userKey)
-            .set("count", 1L)
-            .build();
+        user = Entity.builder(userKey).set("count", 1L).build();
         tx.add(user);
       } else {
         user = Entity.builder(user).set("count", user.getLong("count") + 1L).build();
         tx.update(user);
       }
+      IncompleteKey commentKey = IncompleteKey.builder(userKey, COMMENT_KIND).build();
```

Comment:
All constants in this file (as `COMMENT_KIND`) are those values that users might want to change before running the example (might clash with existing-kinds). Other strings floating are names for entity properties, that won't clash.